x/auctions/types/params_test.go

```diff
@@ -0,0 +1,14 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/evmos/evmos/v19/x/auctions/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParamsValidate(t *testing.T) {
+	params := types.DefaultParams()
+
+	require.NoError(t, params.Validate(), "expected no error with default values")
+}
```

Comment:
I'm really wondering if the whole `params.Validate()` is actually necessary because the functions contained in there are basically just re-checking what static typing does for Go anyways? It's just taking the `bool` then passes that into a function as a `interface{}` only to check if it's a `bool`? That seems totally non-sense

Reply:
Yeah, I discussed the same with @Vvaradinov and the reason is that probably it is possible to bypass the static typing check with binary payload. The Cosmos SDK is using the same approach but would be nice to better understand why this is required